src/wp-includes/option.php

```diff
@@ -86,6 +86,11 @@ function get_option( $option, $default = false ) {
 		return false;
 	}
 
+	static $queried_options = array();
+	if ( isset( $queried_options[ $option ] ) ) {
+		return $queried_options[ $option ];
+	}
```

Comment:
Any form of local caching like this will increase the memory usage of PHP and with a function used as often as this, this could be a significant increase. I'd be interested to see some stats on that.

Secondly, about the code itself - any form of local caching like this would need to take **both** passed parameters into account as otherwise the results will be unstable.

Think:
```php
$my_option = get_option( 'foo' ); // Will return `false` if the option doesn't exist.

$my_option = get_option( 'foo', '' ); // Would previously return '' if the option doesn't exist, would now return `false`.
```

This will also needs additional (unit) tests to be added to verify the correct functioning.

Reply:
Just thinking: a local cache like this may also need to include a hash of the filters hooked into the function.

Depending on where in the code flow the function is called, filters may be hooked in or unhooked, so the results of the function call could be wildly different depending on the filtering being done (not likely, but definitely possible).

And if both the `$default` param + a hash of hooked in filters for all filters used in this function are needed to get this to work correctly, I'm not sure there will be much left of the performance improvement.